LayoutTests/imported/w3c/web-platform-tests/import-maps/dynamic-integrity.html

```diff
@@ -0,0 +1,87 @@
+    "./resources/log.js?pipe=sub&name=InvalidExtra": "sha384-WsKk8nzJFPhk/4pWR4LYoPhEu3xaAc6PdIm4vmqoZVWqEgMYmZgOg9XJKxgD1+8v foobar-rOJN8igD0+jW6lwNN3+InhXTgQztVHlq/HJ0riswXp8kMoiIDx5JpmCwuVem6Ll9q2LFNSu1xq23bsBMMQk1rg==",
+    "./resources/log.js?pipe=sub&name=Suffix": "sha384-lbOWldbmji7sCHI/L8iVJ+elmFIMp41p+aYOLxqQfZMqtoFeHFVe/ASRA0IyZ1/9?foobar",
+    "./resources/log.js?pipe=sub&name=Multiple": "sha384-foobar sha512-rOJN8igD0+jW6lwNN3+InhXTgQztVHlq/HJ0riswXp8kMoiIDx5JpmCwuVem6Ll9q2LFNSu1xq23bsBMMQk1rg==",
+    "./resources/log.js?pipe=sub&name=BadHashWithNoImport": "sha384-foobar",
+    "./resources/log.js?pipe=sub&name=BareURL": "sha384-foobar",
+    "bare2": "sha384-foobar",
+    "resources/log.js?pipe=sub&name=Bare": "sha384-foobar"
+  }
+}
+</script>
+<script type="module">
+const test_not_loaded = (url, description) => {
+  promise_test(async t => {
+    log = [];
+    const promise = import(url);
+    await promise_rejects_js(t, TypeError, promise);
+    assert_array_equals(log, []);
+  }, description);
+};
+
+const test_loaded = (url, log_expectation, description) => {
+  promise_test(async t => {
+    log = [];
+    await import(url);
+    assert_array_equals(log, log_expectation);
+  }, description);
+};
+
+test_not_loaded("./resources/log.js?pipe=sub&name=ResolvesToBadHash",
+  'script was not loaded, as its resolved URL failed its integrity check');
+
+test_loaded("./resources/log.js?pipe=sub&name=ResolvesToNoHash", ["log:NoHash"],
+  'script was loaded, as its resolved URL had no integrity check, despite' +
+  ' its specifier having one');
+
+test_loaded("./resources/log.js?pipe=sub&name=GoodHash", ["log:GoodHash"],
+  'script was loaded, as its integrity check passed');
+
+test_not_loaded("./resources/log.js?pipe=sub&name=BadHashWithNoImport",
+  'Script with no import definition was not loaded, as it failed its' +
+  ' integrity check');
+
+test_not_loaded("bare",
+  'Bare specifier script was not loaded, as it failed its integrity check');
+
+test_loaded("bare2", ["log:F"],
+  'Bare specifier used for integrity loaded, as its definition should have' +
+  ' used the URL');
+
+test_loaded("./resources/log.js?pipe=sub&name=InvalidExtra",
+  ["log:InvalidExtra"],
+  'script was loaded, as its integrity check passed, despite having an extra' +
+  ' invalid hash');
+
+test_loaded("./resources/log.js?pipe=sub&name=Suffix", ["log:Suffix"],
+  'script was loaded, as its integrity check passed, despite having an' +
+  ' invalid suffix');
+
+test_loaded("./resources/log.js?pipe=sub&name=Multiple", ["log:Multiple"],
+  'script was loaded, as its integrity check passed given multiple hashes.' +
+  ' This also makes sure that the larger hash is picked.');
+
+test_loaded("./resources/log.js?pipe=sub&name=Bare",["log:Bare"],
+  'script was loaded, as its integrity check was ignored, as it was defined' +
+  ' using a URL that looks like a bare specifier');
```

Comment:
Nit - can we have this instead? The line breaks/whitepace could end up being a hassle to manually maintain. 

```suggestion
test_not_loaded(
  "./resources/log.js?pipe=sub&name=ResolvesToBadHash",
  "script was not loaded, as its resolved URL failed its integrity check"
);

test_loaded(
  "./resources/log.js?pipe=sub&name=ResolvesToNoHash",
  ["log:NoHash"],
  "script was loaded, as its resolved URL had no integrity check, despite its specifier having one"
);

test_loaded(
  "./resources/log.js?pipe=sub&name=GoodHash",
  ["log:GoodHash"],
  "script was loaded, as its integrity check passed"
);

test_not_loaded(
  "./resources/log.js?pipe=sub&name=BadHashWithNoImport",
  "Script with no import definition was not loaded, as it failed its integrity check"
);

test_not_loaded(
  "bare",
  "Bare specifier script was not loaded, as it failed its integrity check"
);

test_loaded(
  "bare2",
  ["log:F"],
  "Bare specifier used for integrity loaded, as its definition should have used the URL"
);

test_loaded(
  "./resources/log.js?pipe=sub&name=InvalidExtra",
  ["log:InvalidExtra"],
  "script was loaded, as its integrity check passed, despite having an extra invalid hash"
);

test_loaded(
  "./resources/log.js?pipe=sub&name=Suffix",
  ["log:Suffix"],
  "script was loaded, as its integrity check passed, despite having an invalid suffix"
);

test_loaded(
  "./resources/log.js?pipe=sub&name=Multiple",
  ["log:Multiple"],
  "script was loaded, as its integrity check passed given multiple hashes. This also makes sure that the larger hash is picked"
);

test_loaded(
  "./resources/log.js?pipe=sub&name=Bare",
  ["log:Bare"],
  "script was loaded, as its integrity check was ignored, as it was defined using a URL that looks like a bare specifier"
);
```